CenterForOpenScience/osf.io
# [ENG-4360] Feature/keen replacement admin

## Purpose

We're now logging metrics like nightly tallies and pageviews to our internal datastore. We'd like to switch our admin metrics pages over to pull from there instead of Keen. This PR converts the currently supported collections to do that.  We will continue to use the `keen-dataviz` library as our graphing library, so most of this PR is code to remap our metric responses suitable for consumption by `keen-dataviz`.


## Changes

* Create a new admin metrics page (/metrics-mw/) that is a duplicate of the prior one but which pulls from the new metrics store.

* Add new metrics query endpoints for user visits.

* Via @aaxelb: renamed `addon_usage` metric to `storage_addon_usage` and updated it to run in approximately finite time.

Additional changes:

* Added script from @aaxelb to generate test pageview data (osf/management/commands/make_dummy_pageviews_for_metrics.py)
* Fix from @aaxelb to make `daily_reporters_go` properly handle prior dates for backfill purposes.
* Improvement from @aaxelb to exclude reports from `daily_reporters_go`


## QA Notes

* Need to verify that metrics counts are comparable. Start/stop dates may be off-by-one and need to be resolved.

* Make sure NodeLog and download count metrics still pull successfully from Keen. These haven't been ported to the new store yet.

* Test timeframe setting ability of preprint counts page.

* @mfraezz, can you please verify that the nightly addon usage snapshot `osf/metrics/reporters/storage_addon_usage.py` completes in an acceptable amount of time with prod data?


## Documentation

* None, sadly.  This is mostly an internal change.

## Side Effects

None expected

## Ticket

Broad ticket, but: https://openscience.atlassian.net/browse/ENG-3568
